keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -413,28 +412,83 @@ On Pod Create/Update/Delete:
 
 ### Kube-Proxy
 
-Kube-proxy will be modified to consume EndpointSlice instances besides Endpoints resource. A flag will be added to kube-proxy to toggle the mode. 
-
-```
-Watch: Service, EndpointSlice ==> Manage: iptables, ipvs, etc
-```
-- Merge multiple EndpointSlice into an aggregated list.
-- Reuse the existing processing logic 
-
-### Endpoint Controller (classic)
-
-In order to ensure backward compatibility for external consumer of the core/v1 Endpoints API, the existing K8s endpoint controller will keep running until the API is EOL. The following limitations will apply:
-- Starting from EndpointSlice beta: If # of endpoints in one Endpoints object exceed 1000, generate a warning event to the object. 
-- Starting from EndpointSlice GA: Only include up to 1000 endpoints in one Endpoints Object and throw events.
+Kube-Proxy will support consuming EndpointSlice resources as an alternative to
+Endpoints resources. This will be enabled with the `EndpointSliceProxying`
+feature gate.
+
+This will require minimal changes to proxier implementations, instead updating
+the shared code that transforms Endpoints into a generic data structure to also
+transform EndpointSlices into that same data structure.
+
+### Endpoints Controller (classic)
+
+In order to ensure backwards compatibility for external consumer of the core/v1
+Endpoints API, the existing Endpoints controller will keep running until the
+API is EOL. As EndpointSlices become more stable, the Endpoints controller will
```

Comment:
This probably bears explanation. I wouldn't expect any reduction in endpoints functionality until EndpointSlice is GA and part of conformance

Reply:
Right now the roadmap has warning events generated in 1.19 and the first degradation of functionality happening in 1.20, corresponding with the proposed API GA. I think that matches the criteria of functionality not being reduced until GA, but I could clarify that here. Alternatively, I'm open to pushing each of those events back by 1 release. Warnings would then correspond with GA and Endpoints limitations would not be introduced until the release after EndpointSlice GA.

Reply:
I've pushed these events back so they'll now happen as part of the 1.20 and 1.21 release cycle at the earliest. It does seem reasonable to wait until EndpointSlice GA before even introducing warnings on Endpoints.